server/apps/microservices/src/microservices.module.ts

```diff
@@ -0,0 +1,69 @@
+import { BullModule } from '@nestjs/bull';
+import { Module } from '@nestjs/common';
+import { TypeOrmModule } from '@nestjs/typeorm';
+import { DatabaseModule } from '@app/database';
+import { AssetEntity } from '@app/database/entities/asset.entity';
+import { ExifEntity } from '@app/database/entities/exif.entity';
+import { SmartInfoEntity } from '@app/database/entities/smart-info.entity';
+import { UserEntity } from '@app/database/entities/user.entity';
+import { MicroservicesService } from './microservices.service';
+import { AssetUploadedProcessor } from './processors/asset-uploaded.processor';
+import { ThumbnailGeneratorProcessor } from './processors/thumbnail.processor';
+import { MetadataExtractionProcessor } from './processors/metadata-extraction.processor';
+import { VideoTranscodeProcessor } from './processors/video-transcode.processor';
+
+@Module({
+  imports: [
+    DatabaseModule,
+    TypeOrmModule.forFeature([UserEntity, ExifEntity, AssetEntity, SmartInfoEntity]),
+    BullModule.forRootAsync({
+      useFactory: async () => ({
+        redis: {
+          host: process.env.REDIS_HOSTNAME || 'immich_redis',
+          port: 6379,
+        },
+      }),
+    }),
+    BullModule.registerQueue({
+      name: 'thumbnail-generator-queue',
+      defaultJobOptions: {
+        attempts: 3,
+        removeOnComplete: true,
+        removeOnFail: false,
+      },
+    }),
+    BullModule.registerQueue({
+      name: 'asset-uploaded-queue',
+      defaultJobOptions: {
+        attempts: 3,
+        removeOnComplete: true,
+        removeOnFail: false,
+      },
+    }),
+    BullModule.registerQueue({
+      name: 'metadata-extraction-queue',
+      defaultJobOptions: {
+        attempts: 3,
+        removeOnComplete: true,
+        removeOnFail: false,
+      },
+    }),
+    BullModule.registerQueue({
+      name: 'video-conversion-queue',
```

Comment:
There's two definitions for this queue, is that intended?

Reply:
Resolved above.